gradle-plugin/src/main/kotlin/io/spine/protodata/gradle/Plugin.kt

```diff
@@ -0,0 +1,276 @@
+
+    override fun apply(target: Project) {
+        val extension = Extension(target)
+        target.extensions.add("protoData", extension)
+        val protoDataConfiguration = target.configurations.create("protoData")
+        createLaunchTask(target, extension, protoDataConfiguration)
+
+        val resource = Plugin::class.java.classLoader.getResource("version.txt")!!
+        val version = resource.readText()
+        configureProtobufPlugin(target, extension, version)
+
+        val protoDataRawConfiguration = target.configurations.create("protoDataRawArtifact")
+        createInstallTask(target, protoDataRawConfiguration, version)
+    }
+}
+
+/**
+ * The `protoData { }` Gradle extension.
+ */
+@Suppress("UnstableApiUsage") // Gradle Property API.
+public class Extension(private val project: Project) {
+
+    internal  val plugins: ListProperty<String> =
+        project.objects.listProperty(String::class.java)
+
+    /**
+     * Passes given names of Java classes to ProtoData as the `io.spine.protodata.plugin.Plugin`
+     * classes.
+     */
+    public fun plugins(vararg classNames: String) {
+        plugins.addAll(classNames.toList())
+    }
+
+    internal val renderers: ListProperty<String> =
+        project.objects.listProperty(String::class.java)
+
+    /**
+     * Passes given names of Java classes to ProtoData as the `io.spine.protodata.renderer.Renderer`
+     * classes.
+     */
+    public fun renderers(vararg classNames: String) {
+        renderers.addAll(classNames.toList())
+    }
+
+    internal val optionProviders: ListProperty<String> =
+        project.objects.listProperty(String::class.java)
+
+    /**
+     * Passes given names of Java classes to ProtoData as
+     * the `io.spine.protodata.option.OptionsProvider` classes.
+     */
+    public fun optionProviders(vararg classNames: String) {
+        optionProviders.addAll(classNames.toList())
+    }
+
+    internal val requestFileProperty: RegularFileProperty =
+        project.objects.fileProperty().convention(
+            project.layout.buildDirectory.file("protodata/request.bin")
+        )
+
+    /**
+     * The location where to write and read the serialized `CodeGeneratorRequest`.
+     *
+     * The value accepted by this property it turned into a file via the Gradle's `Project.file(..)`
+     * method.
+     *
+     * By default, the request file is stored in a subdir inside the `build` directory.
+     */
+    public var requestFile: Any
+        get() = requestFileProperty.get()
+        set(value) = requestFileProperty.set(project.file(value))
+
+    internal val sourceProperty: DirectoryProperty =
+        project.objects.directoryProperty().convention(
+            project.layout.projectDirectory.dir("generated/main/java")
+        )
+
+    /**
+     * The location where the sources processed by ProtoData can be found.
+     *
+     * By default, `generated/main/java`.
+     */
+    public var source: Any
```

Comment:
Judging on the description, it's rather a destination.

Did you mean "the sources to process by ProtoData"?